liblangutil/Scanner.cpp

```diff
@@ -317,8 +318,12 @@ Token Scanner::scanSingleLineDocComment()
 	{
 		if (tryScanEndOfLine())
 		{
-			// check if next line is also a documentation comment
-			skipWhitespace();
+			// Check if next line is also a documentation comment.
+			// If any whitespaces were skipped, remember source position
+			// and store skipped count.
+			int startPos = m_source->position();
```

Comment:
This feels a bit hacky. Does anyone have another suggestion on how to correct source locations, that always span to the next token (and therefore contains whitespaces)?

Reply:
If you handle setting the end pos in this function, you can just set it at the right place without having to count the whitespace and subtracting it again.

Reply:
Ok, without the usage of the counter, `m_skippedComments[NextNext].location.end ` would now need to be set more then once to cover all cases. I'm not sure if that's better.

Reply:
Either set it multiple times or just store it locally as `endPosition` instead of `whitespaceAppended`.